### ISSUE TITLE
[Bug]: v5.71.0: archive has incorrect │ checksum

### ISSUE THREAD

**Issue by tanayagrawa1** [hashicorp/terraform-provider-aws#39676]
### Terraform Core Version

1.8.5

### AWS Provider Version

5.71.0

### Expected Behavior

pass

### Actual Behavior

```bash
│ Error: Failed to install provider
│ 
│ Error while installing hashicorp/aws v5.71.0: archive has incorrect
│ checksum
│ zh:e746662547a839b0344729783bb57bdf6899f78d3de19e545d58d0acff9567c4
│ (expected
│ zh:c164d3ed13eb4c28ac243221591fc4ac3cfe621f8cd20af533dedd98a6387363)
```

### Terraform Configuration Files

skip

### Steps to Reproduce

install provider

### Would you like to implement a fix?

None

**Comment by joelrozen-bbm** [comment]
+1 all of our workflows are failing with the same error.

**Comment by odcheck** [comment]
│ Error: Failed to install provider│ │ Error while installing hashicorp/aws v5.71.0: archive has incorrect│ checksum│ zh:21f5d77f2859d9ddb5fafa7b6cfd0efdf82977c631ce788e56728afc6da17323│ (expected│ zh:c164d3ed13eb4c28ac243221591fc4ac3cfe621f8cd20af533dedd98a6387363)

**Comment by joelrozen-bbm** [comment]
This issue is intermittent for us - we have ~ 10 CI pipelines, they started producing this error on terraform init this morning, seems that re-running them a few times they will pass and download 5.71 successfully and other times throw the checksum error.

**Comment by gurupreet18121990** [comment]
Hello Team,
Even I am facing the similar issue at present.

09:58:11  │ Error: Failed to install provider
09:58:11  │ 
09:58:11  │ Error while installing hashicorp/aws v5.71.0: archive has incorrect
09:58:11  │ checksum
09:58:11  │ zh:e746662547a839b0344729783bb57bdf6899f78d3de19e545d58d0acff9567c4
09:58:11  │ (expected
09:58:11  │ zh:c164d3ed13eb4c28ac243221591fc4ac3cfe621f8cd20af533dedd98a6387363)
09:58:11  ╵

As a workaround, I am locking the version to 5.70.0 in my provider file. 
  required_providers {
    aws = {
      source  = "hashicorp/aws"
      #version = ">= 3.28"
      version = "= 5.70.0"
    }
  }
}

**Comment by XinRanZhAWS** [comment]
+1, with same wrong checksum, but some time correct

**Comment by nitinjain999** [comment]
`Error: Failed to install provider

Error while installing hashicorp/aws v5.71.0: archive has incorrect checksum
zh:e746662547a839b0344729783bb57bdf6899f78d3de19e545d58d0acff9567c4 (expected
zh:c164d3ed13eb4c28ac243221591fc4ac3cfe621f8cd20af533dedd98a6387363)`

**Comment by jeremychauvet** [comment]
Hello 👋🏼 
Running a `terraform init -upgrade` should help.

**Comment by jeremychauvet** [comment]
![IMG_6845](https://github.com/user-attachments/assets/c9fdc1b9-3cbe-4126-8f2e-2c7c7f5c5bc0)
It seems the tag `5.71.0` has been overridden 2 times, which may explain the checksum issue

**Comment by Guillaume-Alexandre** [comment]
> Hello 👋🏼 Running a `terraform init -upgrade` should help.

It does not work here ... Still the same issue.

**Comment by kevineor** [comment]
The github action looks to have published a version without publish SHA256SUMS
https://github.com/hashicorp/terraform-provider-aws/actions/runs/11282276110/job/31380943498
It have been retried but maybe the terraform registry does still point to the previous provider uploaded

**Comment by andrazjelenc** [comment]
The problem is only with provider for linux amd64 architecture, for other architectures the checksums look fine.

**Comment by ericbl** [comment]
I just lost 2 hours fighting with this issue, I first though about an issue with our plugin cache...

Until the proper checksums get published, the workaround is to force the previous version in your configuration, e.g. with terraform.tf containing

```
terraform {
  required_providers {
    aws = {
      source  = "hashicorp/aws"
      version = "= 5.70.0"
    }
  }
}
```

**Comment by wmenant** [comment]
Tested by adding version = "= 5.70.0" as provided by @gurupreet18121990  and it seems to work on Ubuntu 22.04
![Screen-0273](https://github.com/user-attachments/assets/0297a7f7-4d82-4bde-91e0-a65df08c7339)

**Comment by sunitha8687** [comment]
We have linux/amd64 and tried the workaround suggested above and i still get followup error:

```
    aws = {
      source  = "hashicorp/aws"
      version = "~> 5.45, != 5.71.0"
    }
```
    
<img width="790" alt="image" src="https://github.com/user-attachments/assets/5dfa98ff-dc53-433b-8622-f059dd022d17">

**Comment by ericbl** [comment]
I added a terraform.tf file containing the following along our other rf file, and our ci/cd pipeline is working again.

```
terraform {
  required_providers {
    aws = {
      source  = "hashicorp/aws"
      version = "= 5.70.0"
    }
  }
}
```

but I don't want to add this in our 20+ repos...

**Comment by edmundcraske-bjss** [comment]
> We have linux/amd64 and tried the workaround suggested above and i still get followup error:
> 
> ```
>     aws = {
>       source  = "hashicorp/aws"
>       version = "~> 5.45, != 5.71.0"
>     }
> ```
> 
> <img alt="image" width="790" src="https://private-user-images.githubusercontent.com/44788173/375709172-5dfa98ff-dc53-433b-8622-f059dd022d17.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjg2NDA3NDAsIm5iZiI6MTcyODY0MDQ0MCwicGF0aCI6Ii80NDc4ODE3My8zNzU3MDkxNzItNWRmYTk4ZmYtZGM1My00MzNiLTg2MjItZjA1OWRkMDIyZDE3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDExVDA5NTQwMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTIwMzQ1MDBmMTAyZWZlZGVmNzljMGZjNjlhNWJjNDZlYmJjMGMyMjcxYmMyZmY3ZmU4Yjk2OTBlYTAwNGU1ZjImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.4_oRbOXusXC3najSyKaabVHgXvP0swnGbKRkDAH1jQs">

@sunitha8687 I think you've added it to your provider definition, rather than to your terraform required_providers (as per https://developer.hashicorp.com/terraform/language/providers/requirements)?

**Comment by asktushar** [comment]
Can anyone confirm if there is any ETA for this fix, I don't want to change all the repos in the company.

**Comment by sap147** [comment]
This is causing issues for us as well.

**Comment by stefanodimastromatteo** [comment]
Getting the same error for mac os arch darwin_arm64.

**Comment by rusekjf** [comment]
Hello @austinvalle, is there any update on this bug resolution? Thanks!

**Comment by ymd-xxx** [comment]
This have been fixed! :)

**Comment by anneadb** [comment]
Is the version gone entirely now?

**Comment by justinretzolk** [comment]
Hey everyone 👋 Thank you very much for taking the time to raise this, and for adding reactions to help ensure it was brought to our attention -- that's extremely helpful. We ran into an issue with our release process that led to this, and so have pulled the `5.71.0` release to resolve the immediate issue. We're working through the release issue internally, and will make a new release once we're confident that it has been resolved.

Since the issue reported here has been resolved, I'm going to close this issue out. Thank you again to all who reported -- if you run into issues in the future, please do continue to let us know!

**Comment by rsi-mrobinson** [comment]
anyone else still running into issues like this? Updating my lock files I see 
```
│ Error: Could not retrieve providers for locking
│ 
│ Terraform failed to fetch the requested providers for darwin_amd64 in order to calculate their checksums: some providers could not be installed:
│ - registry.terraform.io/hashicorp/aws: the previously-selected version 5.71.0 is no longer available.
```
Still need the workaround provided above to resolve this.

**Comment by lordmat0** [comment]
@justinretzolk 

Doesn't this go against best practices of deleting versions and instead a new version should be released that rolls back the changes?

Unsure if something like renovate bot can handle these kind of changes.

**Comment by jithinjose2004** [comment]
> @justinretzolk
> 
> Doesn't this go against best practices of deleting versions and instead a new version should be released that rolls back the changes?
> 
> Unsure if something like renovate bot can handle removing a version.

yes 

Error: Resource instance managed by newer provider version
│ 
│ The current state of
│ module.****** was
│ created by a newer provider version than is currently selected. Upgrade the
│ aws provider to work with this state.

**Comment by jithinjose2004** [comment]
@justinretzolk are we going to release a new version anytime soon ?

**Comment by myshon** [comment]
Still not patched... 🤔

**Comment by myshon** [comment]
Could you please release the patch? I would prefer not to modify the pipelines or provider definitions, if possible.

**Comment by justinretzolk** [comment]
@jithinjose2004 thanks for checking in here. We plan on releasing version `5.72.0` on Thursday morning (October 17th), and have updated the changelog to reflect this. 

As far as the best practice of not deleting a version, and instead releasing a new version that rolls back the changes -- in this case, that wasn't really an option. The issue wasn't that there was a particularly offensive bug in the provider; rather that an issue with the release process caused issues with the release itself. Leaving the broken release up would mean that anyone who attempted to install it (most without the context of this GitHub Issue) would receive checksum errors, and never be able to install the release at all. Many people opt not to pin provider versions and so would not be able to use the provider without updating their configurations to explicitly pin a lower version. Queue the influx of high-severity support cases.

The natural next question is as to why we didn't release a new version immediately. Before we could do so, we needed to resolve the issues with the release process -- something we didn't feel we could guarantee on a Friday before a long weekend (half of our team is in Canada, where today is a national holiday). Compounding the difficulty of releasing a new version as quickly as possible, HashiConf, our annual conference [is this week](https://www.hashicorp.com/conferences/hashiconf). With all of those factors in mind, Thursday was the soonest that we were able to get the next release scheduled for. I hope that information helps to give a bit more context as to why we are where we are.
